interpreter/cling/lib/Utils/Output.cpp

```diff
@@ -80,10 +80,12 @@ namespace cling {
 
     bool ColorizeOutput(unsigned Which) {
 #define COLOR_FLAG(Fv, Fn) (Which == 8 ? llvm::sys::Process::Fn() : Which & Fv)
-      return static_cast<ColoredOutput &>(outs()).Colors(
-                              COLOR_FLAG(1, StandardOutIsDisplayed)) |
-             static_cast<ColoredOutput &>(errs()).Colors(
-                              COLOR_FLAG(2, StandardErrIsDisplayed));
+      bool colorStdout = COLOR_FLAG(1, StandardOutIsDisplayed);
+      bool colorStderr = COLOR_FLAG(2, StandardErrIsDisplayed);
+      // The following calls have side effects because they set m_Colorize!
+      static_cast<ColoredOutput &>(outs()).Colors(colorStdout);
+      static_cast<ColoredOutput &>(errs()).Colors(colorStderr);
```

Comment:
Before, the second call might have been shortcut, which makes this a functional change because now `m_Colorize` might be touched for `StandardErrIsDisplayed` where it wasn't touched before. Is that an improvement / bug fix or an unintentional side-effect or not actually happening? Can you call this out in the commit log?

Reply:
No, `|` is not short-circuited because it's a bitwise operation where both operands have to be known. When I replaced with the logical `||`, it was short-circuited and tests started to fail. So actually, this is preserving the status-quo and I added a code comment to prevent somebody clever to come around and "optimize" this in the future.